examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -176,6 +177,83 @@ class _RecipeGridPageState extends State<RecipeGridPage> {
   }
 }
 
+class PestoLogo extends StatefulWidget {
+  PestoLogo({this.showText, this.height});
+
+  final bool showText;
+  final double height;
+
+  @override
+  _PestoLogoState createState() => new _PestoLogoState();
+}
+
+class _PestoLogoState extends State<PestoLogo> {
+  // Native sizes for logo and its image/text components.
+  static const double kLogoHeight = 162.0;
+  static const double kLogoWidth = 220.0;
+  static const double kImageHeight = 108.0;
+  static const double kTextHeight = 48.0;
+  final TextStyle titleStyle = const PestoStyle(fontSize: kTextHeight, fontWeight: FontWeight.w900, color: Colors.white, letterSpacing: 3.0);
+
+  AnimationController _textController;
+  Animation<double> _textOpacity;
+  Animation<double> _imageHeight;
+
+  @override
+  void initState() {
+    super.initState();
+    _textController = new AnimationController(duration: const Duration(milliseconds: 200))
+      ..value = config.showText ? 1.0 : 0.0;
+    _textOpacity = new CurvedAnimation(parent: _textController.view, curve: const Interval(0.3, 1.0, curve: Curves.easeInOut));
+    _imageHeight = new Tween<double>(begin: kLogoHeight, end: kImageHeight).animate(
+      new CurvedAnimation(parent: _textController.view, curve: const Interval(0.0, 0.9, curve: Curves.easeInOut))
+    );
+    _imageHeight.addListener(() {
```

Comment:
This is slick and I'm looking forward to seeing it.

I think it would be a little cleaner to create a Rect animation, like `new RectTween(...).animate(_textController)` and use it with an AnimatedBuilder below. And maybe call _textController just _controller.